viewing_party/party.py

```diff
@@ -1,23 +1,312 @@
+    # return friends_unique_list
+    
+    #ENHANCE CODES FOR READABILITY
+    for friends_dict in friends_watched_list:
+        if friends_dict not in user_watched_list:
+            friends_unique_list.append(friends_dict)
+    # print(friends_unique_list)
+    return friends_unique_list
+
+
+def get_friends_watched_list(user_data):
+    """Helper function to create friends' watched list
+    Input: user_data
+    Output: list of friends watched movie dictionaries
+    """
+    friends_watched_list = []
+
+    if len(user_data["friends"]) == 0:
+        friends_watched_list = []
+    
+    else:
+        # for index1 in range(len(user_data["friends"])): #first loop to give us the movie list (of dictionaries) of "friends"
+        #     for index2 in range(len(user_data["friends"][index1]["watched"])): #second loop give us each element (list of dictionaries of "friends' watched list")
+        #         for index3 in range(len(user_data["friends"][index1]["watched"][index2])): #third loop give us access to each element within the list of friends watched list to compare title
+        #             if user_data["friends"][index1]["watched"][index2] not in friends_watched_list:
+        #                 friends_watched_list.append(user_data["friends"][index1]["watched"][index2])
+        
+        #ENHANCE CODES FOR READABILITY
+        for friend_movie_dicts in (user_data["friends"]): #first loop to give us the movie list (of dictionaries) of "friends"
+
+            for watched_dict in friend_movie_dicts["watched"]: #second loop give us each element (list of dictionaries of "friends' watched list")
+                for key,value in watched_dict.items(): #third loop give us access to each element within the list of friends watched list to compare title
+                    # print(watched_dict["title"])
+                    if watched_dict not in friends_watched_list:
+                        friends_watched_list.append(watched_dict)
         
+        return friends_watched_list
+
+
+def get_user_watched_list(user_data):
+    """Helper function to create user's watch list
+    Input: user data
+    Output: list of user's movie dictionaries
+    
+    """
+    user_watched_list = []
+
+    if len(user_data["friends"]) ==0:
+        user_watched_list = []
+    else:
+        for movie in range(len(user_data["watched"])):
+            user_watched_list.append(user_data["watched"][movie])
+    
+    return user_watched_list
+           
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs(user_data):
+    to_remove_list = get_to_remove_list(user_data)
+
+    user_recs = get_friends_unique_watched(user_data)
+    for element in user_recs[:]:
+        if element in to_remove_list:
+            user_recs.remove(element)
+    return user_recs
+
+
+def get_to_remove_list(user_data):

```

Comment:
The logic in this helper function is doing the exact opposite of what you want the `get_available_recs` function to do. In order to make the list of movies to remove, you're throwing away all the movies you want to keep. Instead of creating the list of things to throw away, make this a list of things you want to keep. On line 211, instead of checking it it's NOT in `user_subscription_list`, if you check that it is, you have a list of potential movies.